packages/block-editor/src/components/link-control/test/index.js

```diff
@@ -1217,10 +1218,9 @@ describe( 'Selecting links', () => {
 		expect( currentLinkHTML ).toEqual(
 			expect.stringContaining( selectedLink.title )
 		);
-		expect( currentLinkHTML ).toEqual(
-			expect.stringContaining( selectedLink.type )
-		);
-		expect( currentLinkHTML ).toEqual( expect.stringContaining( 'Edit' ) );
+		expect(
+			queryByRole( currentLink, 'button', { name: 'Edit' } )
+		).toBeTruthy();
```

Comment:
Fixed a bad test where it's asserting the HTML to contain a `Page` text anywhere. It was passing because the text contains `Hello Page`, but not we want to test against.

Fixed it by using RTL to search for a button with the `Edit` text.